File: common_res/src/main/java/com/yyxnb/common_res/weight/ScaleTransitionPagerTitleView.java
package com.yyxnb.common_res.weight;

import android.content.Context;

import net.lucode.hackware.magicindicator.buildins.commonnavigator.titles.ColorTransitionPagerTitleView;

public class ScaleTransitionPagerTitleView extends ColorTransitionPagerTitleView {

    private float minScale = 0.8f;

    public ScaleTransitionPagerTitleView(Context context) {
        super(context);
    }

    public ScaleTransitionPagerTitleView(Context context, float minScale) {
        super(context);
        this.minScale = minScale;
    }

    @Override
    public void onEnter(int index, int totalCount, float enterPercent, boolean leftToRight) {
        super.onEnter(index, totalCount, enterPercent, leftToRight);
        setScaleX(minScale + (1.0f - minScale) * enterPercent);
        setScaleY(minScale + (1.0f - minScale) * enterPercent);
    }

    @Override
    public void onLeave(int index, int totalCount, float leavePercent, boolean leftToRight) {
        super.onLeave(index, totalCount, leavePercent, leftToRight);
        setScaleX(1.0f + (minScale - 1.0f) * leavePercent);
        setScaleY(1.0f + (minScale - 1.0f) * leavePercent);
    }
}

File: common_base/src/main/java/com/yyxnb/common_base/weight/ScaleTransitionPagerTitleView.kt
package com.yyxnb.common_base.weight

import android.content.Context
import net.lucode.hackware.magicindicator.buildins.commonnavigator.titles.ColorTransitionPagerTitleView

/**
 * 扩展 标题缩放
 */
class ScaleTransitionPagerTitleView(context: Context?) : ColorTransitionPagerTitleView(context) {

    private val minScale = 0.8f

    override fun onEnter(index: Int, totalCount: Int, enterPercent: Float, leftToRight: Boolean) {
        super.onEnter(index, totalCount, enterPercent, leftToRight)
        scaleX = minScale + (1.0f - minScale) * enterPercent
        scaleY = minScale + (1.0f - minScale) * enterPercent
    }

    override fun onLeave(index: Int, totalCount: Int, leavePercent: Float, leftToRight: Boolean) {
        super.onLeave(index, totalCount, leavePercent, leftToRight)
        scaleX = 1.0f + (minScale - 1.0f) * leavePercent
        scaleY = 1.0f + (minScale - 1.0f) * leavePercent
    }
}